lib/fluent/plugin/in_prometheus.rb

```diff
@@ -54,6 +58,8 @@ def configure(conf)
 
       @base_port = @port
       @port += fluentd_worker_id
+
+      raise "Invalid content encoding for the exposed metrics endpoint" unless @content_encoding="identity" || @content_encoding="gzip"
```

Comment:
```suggestion
      raise "Invalid content encoding for the exposed metrics endpoint" unless @content_encoding == "identity" || @content_encoding == "gzip"
```

`@content_encoding="identity"` assigns `identify` to `@content_encoding` and always returns `true`.
So that `@content_encoding` is always `identify` in actual.

Although above suggestion is one of solution, it would be better to change the type of `content_encoding` to `:enum` and remove this line to make it simpler.